doc/devguide/codebase/code-style.rst

```diff
@@ -275,6 +673,40 @@ Goto statements should be used with care. Generally, we use it primarily for err
         return NULL;
     }
 
+Nested goto labels are indented.
+
+.. code-block:: c
+
+    int goto_style_nested()
+    {
+        if (foo()) {
+        label1:
+            bar();
+        }
+
+    label2:
+        return 1;
+    }
+
+TODO - REMOVEME: This is only configurable to left-most as of clang 10 so we
+could just leave the "nested goto" out to "not overload things".
+
+clang-format:
+ - IndentGotoLabels: true (default) [clang10]_
+
+Includes
+~~~~~~~~
+
+TODO
+
+A .c file shall include it's own header first.
+
+TODO - REMOVEME: clang-format could sort includes and group them if configured
```

Comment:
I personally recommend sorting includes for any project I work on, as I just cannot be bothered with that mental drain and it helps cleaning up screwed-up includes that are not standalone, but...

...introducing it to a mature code base requires cleaning up of header files to find missing includes.

It's a one-time thingy to do, typically not a big deal and good for header quality in general, but still it's work. Or you fix them as problems arise, but I personally prefer the "clang-format everything, fix headers, undo clang-format approach, check in header fixes".

Anyhow, that's the reason why I set SortIncludes to false as I am not sure how keen you guys are on that.

Reply:
I would be interested in this as a follow up. But agree that 'false' is the way to go for this round.

Reply:
Will remove the todo for v5